sdk/webpubsub/azure-messaging-webpubsubclient/azure/messaging/webpubsubclient/_client.py

```diff
@@ -206,7 +212,7 @@ def _next_ack_id(self) -> int:
     def _send_message(self, message: WebPubSubMessage, **kwargs: Any) -> None:
         pay_load = self._protocol.write_message(message)
         if not self._ws or not self._ws.sock:
-            raise DisconnectedError("The connection is not connected.")
+            raise SendMessageError("The websocket connection is not connected.")
```

Comment:
Is there a reason we're not using a `WebPubSubConnectionError` or `StartClientError` here?

Reply:
Though if we used an "implicit start" like I mentioned in another comment - perhaps this isn't necessary?

Reply:
> Is there a reason we're not using a `WebPubSubConnectionError` or `StartClientError` here?

Yeah, `WebPubSubConnectionError` is better since it happens when connection is closed.

Reply:
> Though if we used an "implicit start" like I mentioned in another comment - perhaps this isn't necessary?

see reply in https://github.com/Azure/azure-sdk-for-python/pull/33294#discussion_r1423043430

Reply:
Make sense. I think `WebPubSubConnectionError` is better here.